packages/jest-runtime/src/index.ts

```diff
@@ -32,6 +32,7 @@ import {
 import type {V8CoverageResult} from '@jest/test-result';
 import {CoverageInstrumenter, V8Coverage} from 'collect-v8-coverage';
 import * as fs from 'graceful-fs';
+import type ModuleNotFoundError from 'jest-resolve/build/ModuleNotFoundError';
```

Comment:
@SimenB I am not sure about this import, but since `jest-resolve`'s `index.ts` exports using `export = Resolver`, changing that `export default` to then support `export type ...` would break how other modules use `jest-resolve`.

This is what VS Code and Typescript suggested, it is not clean by any means but I don't see any other way of importing the type (not the actual instance of the class, because it is exporte by reference in `static Resolver.ModuleNotFoundError = ModuleNotFoundError`

Reply:
we can export a

```ts
function isModuleNotFoundError(error: Error): error is ModuleNotFoundError {
  return (error as any).code === 'E_MODULE_NOT_FOUND';
}
```

from `jest-resolve` and call that in `jest-runtime` instead of casting and the `.code` check there. Then you don't need to import it at all

Reply:
Good approach, its called `tryCastModuleNotFoundError`